### PR TITLE
arangodb 3.6.2

### PR DESCRIPTION
**Pull request by KVS85**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

@MikeMcQuaid can you please help here? By some mistake or bad will the following things were added into Homebrew recently:
- https://github.com/Homebrew/homebrew-core/pull/50925
- https://github.com/Homebrew/homebrew-core/commit/a8e095ad0a82f1effeaa594f086238526fb4162d#diff-271e832fd581b39f66010ee8f8e18438

That gave wrong absolutely non-official and non-supportable from ArangoDB side version which is being distributed now. It's strange that the following fork was accepted https://github.com/shwang53/homebrew-core/ instead of officially supported by ArangoDB members and community: https://github.com/arangodb-helper/homebrew-core.

We weren't able to produce ArangoDB 3.6.1 before because of dockerless policy violation (https://github.com/Homebrew/homebrew-core/pull/49567), but we did improve things and want to be in good connection with Homebrew!

### CODE REVIEW

**Comment by Bo98** [comment]
> It's strange that the following fork was accepted https://github.com/shwang53/homebrew-core/ instead of officially supported by ArangoDB members and community: https://github.com/arangodb-helper/homebrew-core.

Anyone is allowed to contribute to Homebrew. We don't have a notion of "maintainers" for individual formulae. Equally, we don't mandate that upstream have to directly support formulae, or that people have to have explicit approval for each update from upstream.

That said, there are ways to make sure any update performed is done correctly, such as adding tests that detect if something is wrong.

> That gave wrong absolutely non-official and non-supportable from ArangoDB side version which is being distributed now

Can you clarify this? What precisely about the 3.6.2 update is not supported or official? Is it because of ArangoStarter not being updated?

I never reviewed the original PR (my comment was a mass restart of CIs after an outage). I would say it probably should have used https://download.arangodb.com/Source/ArangoDB-3.6.2.tar.gz over the GitHub autogenerated tarball, but presumably that doesn't matter for ArangoDB (it does for some other projects). However, besides from that can you clarify what exactly broke with the update? Could that be tested for in `brew test` to make sure that doesn't happen again?

> we did improve things and want to be in good connection with Homebrew!

Of course, we appreciate your contributions. This pull request looks good to me.

You should add a `revision 1` between the `sha256` and `head` lines at the top of the formula to make sure people who already have 3.6.2 are offered an update to install with these changes.

**Comment by KVS85** [comment]
> Anyone is allowed to contribute to Homebrew. We don't have a notion of "maintainers" for individual formulae. Equally, we don't mandate that upstream have to directly support formulae, or that people have to have explicit approval for each update from upstream.
> 
> That said, there are ways to make sure any update performed is done correctly, such as adding tests that detect if something is wrong.

Sorry, I thought Homebrew has some kind of off "official" streams like Docker Hub.

> Can you clarify this? What precisely about the 3.6.2 update is not supported or official? Is it because of ArangoStarter not being updated?

Yes. We released new Starter with fixes and would like to avoid any fallback to already fixed issues. Also, real ArangoDB 3.6.2 release isn't the same to what was used.

Otherwise people come to us and Community for help, but we can't (just because we don't maintain this).

> I never reviewed the original PR (my comment was a mass restart of CIs after an outage). I would say it probably should have used https://download.arangodb.com/Source/ArangoDB-3.6.2.tar.gz over the GitHub autogenerated tarball, but maybe that doesn't matter for ArangoDB (EDIT: it does - the sha256 has changed - can you update the URL to the correct one and update the hash?). However, besides from that can you clarify what exactly broke with the update? Could that be tested for in `brew test` to make sure that doesn't happen again?

We will try to avoid this by not providing such tarballs before we submit PR to Homebrew from our side.

Thanks for clarification!

File: Formula/arangodb.rb [hunk 1]
```diff
@@ -12,7 +13,7 @@ class Arangodb < Formula
 
   depends_on "ccache" => :build
   depends_on "cmake" => :build
-  depends_on "go" => :build
+  depends_on "go@1.13" => :build
```
Comment (MikeMcQuaid) [inline Formula/arangodb.rb]: @KVS85 Does this fail to build with `go`?

**Comment by MikeMcQuaid** [comment]
> We will try to avoid this by not providing such tarballs before we submit PR to Homebrew from our side.

That would be great. If you can suggest any other way of automatically detecting this that would be good e.g. something in the `test do` block. Thanks!

File: Formula/arangodb.rb (hunk 1 above)
Comment (KVS85) [reply Formula/arangodb.rb]: For some reason - yes. Jenkins output stated "Go 1.13 required" before (in Catalina and Mojave which we only support from our side). Maybe it's just a matter of image and what "go" is there (should be 1.14 as far as I can see).

Comment (KVS85) [reply Formula/arangodb.rb]: We didn't test it with 1.14.

Comment (MikeMcQuaid) [reply Formula/arangodb.rb]: If it fails without on our Jenkins: no problem 👍. Just note in future we may not always have the version of your choice available.

**Pull request closed without merging**